frame.js

```diff
@@ -0,0 +1,70 @@
+class Frame {
+  constructor(round = 1) {
+    this.points = 0;
+    this.round = round;
+    this.rollOne = null;
+    this.rollTwo = null;
+    this.rollThree = null;
+    this.complete = false;
+  };
+
+  roll(pins) {
+    if (this.complete) throw "It's over! Stop lobbing the balls!";
+  
+    if (this.rollOne === null) {
+      this.firstGo(pins);
+    } else if (this.round < 10 && this.rollTwo === null) {
+      this.secondGo(pins);
+    } else if (this.rollTwo === null) {
+      this.secondGoRound10(pins);
+    } else {
+      this.thirdGo(pins);
+    }
+  }
```

Comment:
So if I'm reading this correctly - it is the job of the Frame class currently to assign the incoming roll to the correct part of the frame - using the Frame methods firstGo, secondGo, etc.?